Facebooky/ProxyDataSource.cs

```diff
@@ -0,0 +1,233 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Windows.Forms;
+using FacebookWrapper.ObjectModel;
+
+namespace Facebooky
+{
+	class ProxyDataSource : Component, IBindingListView
+	{
+		private List<Post> m_FilteredPosts;
+		private User m_User;
+		public bool ProxyEnabled { get; set; }
+		private ICollection<PostFilterGroup> m_PostFilterGroups;
+		private readonly BindingSource r_OriginalBindingSource;
+
+		public ProxyDataSource(User i_User, ICollection<PostFilterGroup> i_FilterGroup)
+		{
+			ProxyEnabled = true;
+			this.r_OriginalBindingSource = new BindingSource(i_User, "NewsFeed");
+			m_User = i_User;
+			m_PostFilterGroups = i_FilterGroup;
+			m_FilteredPosts = new List<Post>();
+			this.filterPosts();
+			this.r_OriginalBindingSource.ListChanged += this.DoWhenListChange;
+			//To Saggi- I get an error about syncroot. Do you think I should forward it to the syncroot of the original databinding? or to creat a new one? No one references it.
+		}
+		public void DoWhenListChange(object i_Sender, EventArgs i_Args)
+		{
+			this.filterPosts();
+		}
+		private void filterPosts()
+		{
+			this.m_FilteredPosts = this.m_User.NewsFeed.Where(i_Post => this.getPostPriority(i_Post) != ePostPriority.Hidden).ToList();
+			this.m_FilteredPosts.Sort(
+			(i_Post, i_OtherPost) => this.getPostPriority(i_OtherPost).CompareTo(this.getPostPriority(i_Post)));
+		}
+		private ePostPriority getPostPriority(Post i_Post)
```

Comment:
I know that I originally name the function but please change it to calculatePostPriority() since it's more descriptive.